autotest/gcore/vrt_read.py

```diff
@@ -1620,10 +1622,10 @@ def test_vrt_protocol():
     assert ds.GetRasterBand(1).YSize == 10
 
     ## separated oo instances not allowed
-    with pytest.raises(Exception):
-        assert not gdal.Open(
-            "vrt://data/byte_with_ovr.tif?oo=GEOREF_SOURCES=TABFILE&oo=OVERVIEW_LEVEL=0"
-        )
+    # with pytest.raises(Exception, match="option should be specified once"):
+    #     gdal.Open(
+    #         "vrt://data/byte_with_ovr.tif?oo=GEOREF_SOURCES=TABFILE&oo=OVERVIEW_LEVEL=0"
```

Comment:
`VRTDataset::OpenVRTProtocol` is using `CSLTokenizeString2` to parse the query string. It looks like this function discards the second `oo` that is expected to produce an error here.

Reply:
No, CSLTokenizeString2() does the job properly, but there were missing return nullptr. Fixed per e7cf790a2d2c1d6de37d77c9d474018bfcb376c8. Rebasing on top of latest master should fix it

Reply:
Done